### PR TITLE
Added PinItemWidgetBuilder

### PR DESCRIPTION
**Pull request by cogivn**
Create a PinItemWidgetBuilder to help customize pin item

### CODE REVIEW

File: lib/src/widgets/_pin_item.dart [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 class _PinItem extends StatelessWidget {
   final _PinputState state;
   final int index;
+  final PinItemWidgetBuilder? builder;
 
-  const _PinItem({required this.state, required this.index});
+  const _PinItem({
+    required this.state,
+    required this.index,
+    this.builder,
```
Comment (Tkko) [inline lib/src/widgets/_pin_item.dart:13]: Let's make builder required as well, we are always passing it.

**Comment by enghoelee** [comment]
are we merging this soon?

File: lib/src/widgets/_pin_item.dart [hunk 2]
```diff
@@ -1,10 +1,17 @@
 part of '../pinput.dart';
 
+typedef PinItemWidgetBuilder = Widget Function(String, PinTheme);
```
Comment (enghoelee) [inline lib/src/widgets/_pin_item.dart:3]: Can we include the index in as well?

**Comment by Tkko** [comment]
Hey, I will make some changes and add this feature in the next release

**Pull request merged**